packages/core/src/internal/effectRunnerMap.js

```diff
@@ -37,10 +37,17 @@ function createTaskIterator({ context, fn, args }) {
       return result
     }
 
-    const next = (value = result) => ({
```

Comment:
If the promise resolves to be `undefined`, next time `next` is called with the resolved value, the default argument `result` is used.